x-pack/platform/packages/shared/kbn-inference-endpoint-ui-common/src/components/additional_options_fields.tsx

```diff
@@ -175,6 +175,102 @@ export const AdditionalOptionsFields: React.FC<AdditionalOptionsFieldsProps> = (
     ]
   );
 
+  const temperatureSettings = useMemo(
+    () =>
+      taskTypeOptions?.some((option) => option.id === CHAT_COMPLETION_TASK_TYPE) ||
```

Comment:
Since this is only compatible with chat completion, then we should enforce that in the UI by only showing this setting if chat completion is selected.